sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -80,14 +137,70 @@ internal class NextLinkOperationImplementation : IOperation
             OperationFinalStateVia finalStateVia,
             string? apiVersion)
         {
-            _requestMethod = requestMethod;
+            AssertNotNull(pipeline, nameof(pipeline));
+            AssertNotNull(requestMethod, nameof(requestMethod));
+            AssertNotNull(startRequestUri, nameof(startRequestUri));
+            AssertNotNull(nextRequestUri, nameof(nextRequestUri));
+            AssertNotNull(headerSource, nameof(headerSource));
+            AssertNotNull(finalStateVia, nameof(finalStateVia));
+
+            RequestMethod = requestMethod;
             _headerSource = headerSource;
             _startRequestUri = startRequestUri;
             _nextRequestUri = nextRequestUri;
             _lastKnownLocation = lastKnownLocation;
             _finalStateVia = finalStateVia;
             _pipeline = pipeline;
             _apiVersion = apiVersion;
+            OperationId = ParseOperationId(nextRequestUri);
+        }
+
+        private static string? ParseOperationId(string nextRequestUri)
+        {
+            if (Uri.TryCreate(nextRequestUri, UriKind.Absolute, out var uri))
+            {
+                return uri.Segments.LastOrDefault();
+            }
+            return null;
+        }
+
+        public RehydrationToken GetRehydrationToken()
+            => GetRehydrationToken(RequestMethod, _startRequestUri, _nextRequestUri, _headerSource.ToString(), _lastKnownLocation, _finalStateVia.ToString(), OperationId);
+
+        public static RehydrationToken GetRehydrationToken(
+            RequestMethod requestMethod,
+            Uri startRequestUri,
+            Response response,
+            OperationFinalStateVia finalStateVia,
+            bool skipApiVersionOverride = false,
+            string? apiVersionOverrideValue = null)
+        {
+            string? apiVersionStr;
+            if (apiVersionOverrideValue is not null)
+            {
+                apiVersionStr = apiVersionOverrideValue;
+            }
+            else
+            {
+                apiVersionStr = !skipApiVersionOverride && TryGetApiVersion(startRequestUri, out ReadOnlySpan<char> apiVersion) ? apiVersion.ToString() : null;
+            }
+            var headerSource = GetHeaderSource(requestMethod, startRequestUri, response, apiVersionStr, out var nextRequestUri);
+            response.Headers.TryGetValue("Location", out var lastKnownLocation);
```

Comment:
What happens if this returns false?

Reply:
The logic exists [here](https://github.com/Azure/azure-sdk-for-net/pull/42686/files/adc92770313b373c0e0ed3afd642114d21b1b967#diff-ddf89b49baf148b40544d15990fb244dfa02c26a8a67da24903cb6af0b023e1cR61), we need to duplicate it to create the rehydration token.

Reply:
It would be better to check the return value of the TryGetValue and explicitly set `lastKnownLocation` to null if it returns false.  The use of `var` here hides the fact that it is nullable, and in general the semantics of TryGet methods are that if they return false, the value of the `out` variable is not defined -- it could be anything.

Reply:
sounds good, updated